src/func/imports/delete-import.js

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2018 Shivam Tripathi
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+// @flow
+
+import type {Transaction} from '../types';
+
+
+export default async function deleteImport(
+	transacting: Transaction, importId: number
+) {
+	// Get the type of the import
+	const importType = await transacting.select('type')
```

Comment:
I think you need to consider the order of deletion operations here more carefully. The vote and header deletes can both happen in parallel. As soon as those are both done, the import delete can happen. As soon as the header is deleted, the data delete can happen.

So something more like:

    const headerDeletePromise = ...

    const dataDeletePromise = headerDeletePromise.then(...)

    await Promise.all([
        headerDeletePromise,
        transacting('bookbrainz.discard_votes')...
    ])

    await transacting('bookbrainz.import')...